src/python/dxpy/cli/download.py

```diff
@@ -169,7 +169,7 @@ def download(args):
 
         matching_folders = []
         # project may be none if path is an ID and there is no project context
-        if project is not None:
+        if project is not None and folderpath is not None:
```

Comment:
We initialize some variables in this block (e.g. `strip_prefix `) that are used later. It seems to be guarded though by errors thrown in lines 185 and 198, so it looks okay to me just not easy to read.